lib/plugins/package/lib/packageService.js

```diff
@@ -59,44 +61,115 @@ module.exports = {
       });
   },
 
+  // Inspired by https://github.com/rxaviers/async-pool
+  // maps an array over an interator function that returns an array of promises
+  // with the number of concurrent iterations limited to `limit`
+  _asyncPool(array, fn, limit) {
```

Comment:
This looks like a candidate for a generic tool that can be used in other places as well e.g. for uploads, so I would recommend to put it in a separate module where it can be reused